compiler/resolution/functionResolution.cpp

```diff
@@ -1664,27 +1825,55 @@ handleSymExprInExpandVarArgs(FnSymbol*  workingFn,
         newFormal->cname        = astr("_e", istr(i), "_", formal->cname);
 
         tupleCall->insertAtTail(new SymExpr(newFormal));
-
         formal->defPoint->insertBefore(newArgDef);
+        varargFormals[i] = newFormal;
       }
 
+      bool needTupleInBody;
+
       // Replace mappings to the old formal with mappings to the new variable.
       if (workingFn->hasFlag(FLAG_PARTIAL_COPY)) {
+        bool gotFormal = false; // for assertion only
```

Comment:
Yes

On May 11, 2016, at 1:32 PM, Vassily Litvinov <notifications@github.com<mailto:notifications@github.com>> wrote:


In compiler/resolution/functionResolution.cpp<https://github.com/chapel-lang/chapel/pull/3838#discussion_r62921196>:

>        // Replace mappings to the old formal with mappings to the new variable.
>        if (workingFn->hasFlag(FLAG_PARTIAL_COPY)) {
> +        bool gotFormal = false; // for assertion only


Given that this is pre-existing code, do you insist on fixing the issue that you point out?

—
You are receiving this because you were mentioned.
Reply to this email directly or view it on GitHub<https://github.com/chapel-lang/chapel/pull/3838/files/20b76cba1018c724ee2d739174a99cfcfbb22a17#r62921196>